framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1079,6 +1084,43 @@ protected void printOrStoreMessage(
         Trees.instance(processingEnv).printMessage(kind, message, source, root);
     }
 
+    /**
+     * printOrStoreMessage method with an added stack trace argument. The stack trace is printed
+     * when the dumpOnErrors option is enabled.
+     *
+     * @param kind the kind of message to print
+     * @param message the message text
+     * @param source the souce code position of the diagnostic message
+     * @param root the compilation unit
+     * @param trace the stack trace where the checker encountered an error
+     */
+    protected void printOrStoreMessage(
+            javax.tools.Diagnostic.Kind kind,
+            String message,
+            Tree source,
+            CompilationUnitTree root,
+            StackTraceElement[] trace) {
+        Trees.instance(processingEnv).printMessage(kind, message, source, root);
```

Comment:
Add an implementation comment why here you don't just call the other `printOrStoreMessage` overload.

Reply:
We can alternatively call the other ```printOrStoreMessage``` and it wouldn't affect the output. So I don't have a particular reason for not calling the function. If you think I should call the function, I can make a commit accordingly.

Reply:
I don't think that would give the same behavior, because `BaseTypeChecker` overrides that method and you would get two stack traces, wouldn't you?

Reply:
I called the other function in the ```printOrStoreMessage``` method. It printed the stack trace once but didn't print the error itself. I have added an implementation comment about this behaviour in my latest commit.